x-pack/platform/plugins/private/data_visualizer/public/lite/index_input.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFieldText, EuiFormRow, EuiSpacer, EuiTitle } from '@elastic/eui';
+import type { FC } from 'react';
+import React, { useState } from 'react';
+import useDebounce from 'react-use/lib/useDebounce';
+import type { FileUploadStartApi } from '@kbn/file-upload-plugin/public/api';
+import { i18n } from '@kbn/i18n';
+import { STATUS } from './file_manager/file_manager';
+
+interface Props {
+  setIndexName: (name: string) => void;
+  setIndexValidationStatus: (status: STATUS) => void;
+  fileUpload: FileUploadStartApi;
+}
+
+export const IndexInput: FC<Props> = ({ setIndexName, setIndexValidationStatus, fileUpload }) => {
+  const [indexNameLocal, setIndexNameLocal] = useState('');
+  const [indexNameError, setIndexNameError] = useState('');
+
+  useDebounce(
+    async () => {
+      setIndexValidationStatus(STATUS.STARTED);
+      if (indexNameLocal === '') {
+        setIndexValidationStatus(STATUS.COMPLETED);
+        setIndexNameError('');
+        setIndexName('');
+        return;
+      }
+      const exists = await fileUpload.checkIndexExists(indexNameLocal);
+      const error = exists
+        ? i18n.translate(
+            'xpack.dataVisualizer.file.importView.indexNameAlreadyExistsErrorMessage',
+            {
+              defaultMessage: 'Index name already exists',
+            }
+          )
+        : isIndexNameValid(indexNameLocal);
+      setIndexName(indexNameLocal);
```

Comment:
as it is an async function, there is a change of setting a state for unmounted component.

Reply:
Good spot

Reply:
Updated in https://github.com/elastic/kibana/pull/206864/commits/8bf59f8a149da53e5f73aa55cf472bf9df13c11b